client_cli/src/main.rs

```diff
@@ -892,6 +906,130 @@ mod asset {
+            Ok(())
+        }
+    }
+    #[derive(StructOpt, Debug)]
+    pub struct RemoveStore {
+        #[structopt(long = "asset-id")]
+        pub asset_id: AssetId,
+        /// The key for the store value
+        #[structopt(long)]
+        pub key: Name,
+    }
+
+    impl RunArgs for RemoveStore {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            let Self { asset_id, key } = self;
+            let remove = RemoveKeyValue::asset(asset_id, key);
+            submit([remove], UnlimitedMetadata::default(), context)?;
+            Ok(())
+        }
+    }
+
+    #[derive(StructOpt, Debug)]
+    pub struct GetStore {
+        #[structopt(long = "asset-id")]
+        pub asset_id: AssetId,
+        // The key for the store value
+        #[structopt(long)]
+        pub key: Name,
+    }
+
+    impl RunArgs for GetStore {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            let Self { asset_id, key } = self;
+            let value = get_store(asset_id, &key, context)?;
+            context.print_data(&value)?;
+            Ok(())
+        }
+    }
+
+    #[derive(StructOpt, Debug)]
+    pub struct UpdateStore {
+        /// AssetId for the Store asset (in form of `asset##account@domain_name')
+        #[structopt(long = "asset-id")]
+        pub asset_id: AssetId,
+        /// The key for the store value
+        #[structopt(long)]
+        pub key: Name,
+        /// The new value to be associated with the key
+        #[structopt(long)]
+        pub value: String,
+        /// Interpret the `value` as a JSON string
+        #[structopt(long = "json", requires = "value")]
+        pub is_json: bool,
+    }
+
+    impl RunArgs for UpdateStore {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            let Self {
+                asset_id,
+                key,
+                value,
+                is_json,
+            } = self;
+
+            // Verify the key exists
+            let _value_exists = get_store(asset_id.clone(), &key, context)?;
+            let v: Value = if is_json {
+                serde_json::from_str(&value)?
+            } else {
+                value.parse()?
+            };
+
+            let update = SetKeyValue::asset(asset_id, key, v);
+            submit([update], UnlimitedMetadata::default(), context)?;
+            Ok(())
+        }
+    }
+
+    fn get_store(asset_id: AssetId, key: &Name, context: &dyn RunContext) -> Result<JsonValue> {
+        let client = Client::new(context.configuration())?;
+        let asset = client
+            .request(asset::by_id(asset_id))
+            .wrap_err("Failed to get store asset")?;
+        let get_json = json!(&asset);
+        let value = &get_json["value"]["Store"][key.to_string()];
+        match value {
+            serde_json::Value::Null => Err(eyre!("Key '{}' not found in the store.", key)),
+            _ => Ok(value.clone()),
+        }
+    }
```

Comment:
You can use `FindAssetKeyValueByIdAndKey` query to get value directly

Reply:
And I think _should_, because this query is designed specifically for that. Store assets might have a lot of entries.